src/capture/image/png_writer.cpp

```diff
@@ -206,19 +206,22 @@ void PngWriter::WritePngInfo(const uint16_t width, const uint16_t height,
 
 	char software_keyword[] = "Software";
 	static_assert(sizeof(software_keyword) < 80, "libpng limit");
-	char software_value[] = CANONICAL_PROJECT_NAME " " VERSION;
+
+	auto software_value = format_string("%s %s",
+	                                    CANONICAL_PROJECT_NAME,
+	                                    DOSBOX_GetDetailedVersion());
 
 	texts[0].compression = PNG_TEXT_COMPRESSION_NONE;
 	texts[0].key         = static_cast<png_charp>(software_keyword);
-	texts[0].text        = static_cast<png_charp>(software_value);
+	texts[0].text        = static_cast<png_charp>(software_value.data());
```

Comment:
More of this `.data()` vs `.c_str()` mess where the standard doesn't guarantee the former will be null terminated.  It looks like `format_string` is a new C++ 20 feature that returns `string_view` which is somewhat problematic when we need a C style string.

Reply:
@weirddan455 Citation needed, `std::format` returns `std::string` according to this:
https://en.cppreference.com/w/cpp/utility/format/format

Reply:
>  Class template std::basic_format_string wraps a [std::basic_string_view](https://en.cppreference.com/w/cpp/string/basic_string_view) that will be used by formatting functions.

https://en.cppreference.com/w/cpp/utility/format/basic_format_string

Maybe that's not the return value.  These templated things are hard to read.  If you get a `std::string` back, you should be able to use `.c_str()` and avoid any ambiguity.

Reply:
> > Class template std::basic_format_string wraps a [std::basic_string_view](https://en.cppreference.com/w/cpp/string/basic_string_view) that will be used by formatting functions.
> 
> https://en.cppreference.com/w/cpp/utility/format/basic_format_string
> 
> Maybe that's not the return value. These templated things are hard to read. If you get a `std::string` back, you should be able to use `.c_str()` and avoid any ambiguity.

Not in this case because `c_str()` returns a `const char *` but `data()` returns a `char *`. Then PVS-Studio complains about casting the `const`ness away, which I'd need to do to pass it to the C API that wants a `char *`.

When you have a `std::string`, using `data()` is a good solution as that returns a `char *`, without the `const`, and that's what I'm doing here.